packages/scenario/test/serialization.test.ts

```diff
@@ -116,3 +117,121 @@ describe('SameRefDifferentInstancesIssue449Test.java (regression tests)', () =>
+		 *    upon deserialization it will once again attempt to retrieve the
+		 *    external instance resource
+		 * 5. While parsing will produce a "placeholder" for an unretrievable
+		 *    external instance resource, deserialization will produce an error
+		 *    instead.
+		 * 6. It is expected that a client will resolve this error condition by
+		 *    bypassing deserialization (and potentially discarding the serialized
+		 *    state?), by re-parsing the form... thereby allowing JavaRosa to once
+		 *    again produce a "placeholder".
+		 *
+		 * It's unclear if any of this is pertinent to web forms, but it's hard not
+		 * to ask... if this mental model is roughly correct:
+		 *
+		 * - Is this ceremony something essential to the distinction between "parse"
+		 *   and "deserialize"? Is there additional mental model to help make the
+		 *   distinction more clear?
+		 *
+		 * - Insofar as we may find ourselves implementing similar logic (albeit
+		 *   serving other purposes), how can we establish a clear interface
+		 *   contract around behaviors like this? Should it be more consistent? Does
+		 *   our current {@link EngineConfig.fetchResource} option—configurable in
+		 *   {@link InitializeFormOptions}—provide enough informational surface area
+		 *   to communicate such intent (and allow both clients and engine alike to
+		 *   have clarity of that intent at call/handling sites)?
+		 *
+		 * - - -
+		 *
+		 * JR:
+		 *
+		 * Clients would typically catch this exception and try parsing the form
+		 * again which would succeed by using the placeholder.
+		 */
+		it.skip(
+			'fileNotFoundException_whenFormIsDeserialized_afterPlaceholderInstanceUsed_andFileStillMissing'
+		);
+
+		/**
+		 * **PORTING NOTES**
+		 *
+		 * Expanding on the mental model we're trying to form in the skipped test
+		 * directly above...
+		 *
+		 * - If this produces an error condition, why is it deferred to _accessing a
+		 *   select's available itemset choices_?
+		 *
+		 * - The JavaRosa comment, preserved below, suggests that the error should
+		 *   occur "when making a choice". That makes it sound like the intent is:
+		 *
+		 *     0. Parsing fails to resolve an external instance resource, producing
+		 *        a "placeholder" representation.
+		 *
+		 *     1. Deserialization resolves that resource.
+		 *
+		 *     2. This is considered an inconsistent state, which cannot be
+		 *        reconciled on the deserialized state.
+		 *
+		 *     3. An allowance is made for operating on form state with this
+		 *        inconsistency **so long as the inconsistency isn't consulted in a
+		 *        subsequent state change**.
+		 *
+		 *     If that's the case... why isn't the error produced on a write call,
+		 *     rather than when attempting to read the options? Wouldn't this
+		 *     produce an error by simply proceeding through the deserialized form
+		 *     state without making any state change at all?
```

Comment:
Thanks! I'm quoting this entire comment verbatim in situ, as it's super helpful context. I'll also call out here:

> In some cases, this can result in a significant perf improvement.

This is something we could probably think about in Web Forms too, when we have space to think about such things. At this point, everything to do with itemsets is my top performance optimization candidate, and I suspect that won't change much between here and the rest of the major engine functionality.

I think it'd be relatively low impact in terms of the engine/client boundary, as select choices are already exposed as a computed-read accessor. I suspect the engine-side optimization may already be part of the way there just by virtue of using derived computations; the rest of the work would be deferring initial compute (on load, on any internal code path that isn't invoked by the client reading state).

The biggest other chunk of work would probably be client side, where:

1. In a full-form-view presentation, the opportunity for laziness/deferred reads would generally be concerned with scroll position (or any other means of determining what's "off screen").
2. We'll really want to think about how laziness in such a presentation would affect UX. We don't want the form view and questions jumping around when triggering compute of nearby/previously off-screen choices. (This is also much less of a concern with fixed-size controls, and much more of a concern with e.g. checkbox/radio lists or similar appearances.)